Sources/TuistCore/Graph/XCFrameworkNode.swift

```diff
@@ -0,0 +1,65 @@
+import Basic
+import Foundation
+import TuistSupport
+
+final class XCFrameworkNode: PrecompiledNode {
+    let libraries: [XCFrameworkInfoPlist.Library]
+
+    init(path: AbsolutePath, libraries: [XCFrameworkInfoPlist.Library]) {
+        self.libraries = libraries
+        super.init(path: path)
+    }
+
+    static func parse(
+        path: AbsolutePath,
+        cache: GraphLoaderCaching,
+        configurationFile: String = "Info.plist",
+        fileHandler: FileHandling = FileHandler.shared
+    ) throws -> XCFrameworkNode {
+        if let xcframeworkNode = cache.precompiledNode(path) as? XCFrameworkNode {
+            return xcframeworkNode
+        }
+        
+        let infoPlist = path.appending(component: configurationFile)
+        guard fileHandler.exists(infoPlist) else {
+            throw GraphLoadingError.missingFile(infoPlist)
+        }
+        
+        let config: XCFrameworkInfoPlist = try fileHandler.readPlistFile(infoPlist)
+        // TODO: implement library info.plist parsing
+        let xcframeworkNode = XCFrameworkNode(path: path, libraries: config.libraries)
+        cache.add(precompiledNode: xcframeworkNode)
+        return xcframeworkNode
+    }
+
+    override var binaryPath: AbsolutePath {
+        return XCFrameworkNode.binaryPath(frameworkPath: path, libraries: libraries)
+    }
+	
+    override func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(path.pathString, forKey: .path)
+        try container.encode(name, forKey: .name)
+        try container.encode("precompiled", forKey: .type)
+    }
+	
+    /// Given a framework path and libraries it returns the path to its binary.
+    /// - Parameter frameworkPath: Framework path.
+    /// - Parameter libraries: Framework available libraries
+    static func binaryPath(
+        frameworkPath: AbsolutePath,
+        libraries: [XCFrameworkInfoPlist.Library]
+    ) -> AbsolutePath {
+        let library = libraries.first { $0.architectures.contains(.arm64) }!
```

Comment:
We can obtain the first binary here - is there a reason we are specifically looking for `.arm64`?

xcframeworks aren't guaranteed to contain that architecture.

Reply:
Only `ios-x86_64-simulator` and `ios-arm64` have respective binaries in 
`**{FRAMEWORK-NAME}**.framework\**{FRAMEWORK-NAME}**`

but `ios-x86_64-maccatalyst` has it in deep nested level folders as following
**{FRAMEWORK-NAME}**.framework\Versions\A\**{FRAMEWORK-NAME}**

I think, we also need to consider **static** libs